Add OSO Energy sensor documentation (#30889)
home-assistant/home-assistant.github.io: source/_integrations/osoenergy.markdown

## Platforms

### Sensor

The `osoenergy` sensor integration exposes OSO Energy data as a sensor.

The platform exposes the following sensors:

- Heater Mode for water heaters.
- Optimization Mode for water heaters.
- Power load (kW) for water heaters.
- Tapping capacity (kWh) for water heaters.
- Capacity mixed water at 40°C (L) for water heaters.

> - V40 Min (L) for water heaters.

Reviewer: what is V40 Min (L)? What does this do?

Author: V40 Min (L) is the minimum allowed quantity of mixed hotwater at 40 degrees C for the heater. 
It serves as a guard to turn on the heating if the temperature of the water drops below that treshold - when the capacity drops below the v40 min.
This guard can change depending on the operations on the water heater. For example it can drop to 0 is a user activates sleep mode.

Reviewer: Thank you for the explanation.
So the first few words before "for water heater" is the actual name of the sensor? Could you maybe add a brief explanation for each sensor, like you just did? I think this would improve the value of the documentation. At least for me, the names alone don't say it all.

Author: I have added descriptions to the sensors.

Reviewer: Nice :star_struck: Thank you!

> ## Platforms

Reviewer: ```suggestion
```
Not related to your PR, but we stopped referring to platforms in user documentation.

> ### Sensor

Reviewer: ```suggestion
## Sensors
```

> The `osoenergy` sensor integration exposes OSO Energy data as a sensor.

Reviewer: ```suggestion
The **OSO Energy** integration exposes OSO Energy data as a sensor. It provides the following sensors:
```

> 
> The platform exposes the following sensors:

Reviewer: ```suggestion
```

- Heater Mode for water heaters.
  - Indication of the current heater mode on the device.
- Optimization Mode for water heaters.

>   - Indication of the way heating is managed by the device according to external factors, e.g. electricity prices.

Reviewer: ```suggestion
  - Indication of the way heating is managed by the device according to external factors. For example, electricity prices.
```
Changed to follow the [MS Style Guide]

- Power load (kW) for water heaters.
  - Indication of the current power load of the water heater - how much energy is currently drawn out of the electricity grid.
- Tapping capacity (kWh) for water heaters.
  - Indication for quantity of energy that is stored and available for use at the moment - the current kWh equivalent capacity of the water heater.
- Capacity mixed water at 40°C (L) for water heaters.
  - Indication of the current quantity of water that can be drained out after mixing the water in the heater with the inlet tap water to produce water at 40°C.
- V40 Min (L) for water heaters.

>   - The current minimum allowed quantity of mixed water at 40°C. When the current capacity drops bellow this threshold the heater will turn on as a feature to guard against cold water.

Reviewer: ```suggestion
  - The current minimum allowed quantity of mixed water at 40°C. When the current capacity drops below this threshold, the heater will turn on as a feature to guard against cold water.
```

- Minimum Level of V40 Min (L) for water heaters.

>   - The lower boundary of the V40 Min sensor that can be achieved during normal work of the water heater. Can vary depending on the type of the heater.

Reviewer: ```suggestion
  - The lower boundary of the V40 Min sensor that can be achieved during normal operation of the water heater. Can vary depending on the type of the heater.
```

- Maximum Level of V40 Min (L) for water heaters.

>   - The upper boundary of the V40 Min sensor that can be achieved during normal work of the water heater. Can vary depending on the type of the heater.

Reviewer: ```suggestion
  - The upper boundary of the V40 Min sensor that can be achieved during normal operation of the water heater. Can vary depending on the type of the heater.
```

> ### Water Heater

Reviewer: ```suggestion
## Water heater
```

> The OSO Energy water heater platform integrates your OSO Energy devices into Home Assistant.

Reviewer: ```suggestion
The OSO Energy water heater integration integrates your OSO Energy devices into Home Assistant.
```

> The platform supports the following OSO Energy devices:

Reviewer: ```suggestion
It supports the following OSO Energy devices:
```
